### PR TITLE
URI and Address of Record parsing

### PR DESCRIPTION
**Pull request by audiolion**
implements parsers for URI and AOR (Address of Record), addresses are found typically in the `from, to, and via` fields

### CODE REVIEW

File: src/parser/to-js/header.ts [hunk 1]
```diff
@@ -5,13 +5,15 @@
   header, delimited by ";".
 */
 
+const re = /=(.+)/;
```
Comment (zackify) [inline src/parser/to-js/header.ts]: why a variable for this?

File: src/parser/to-js/uri.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+const re = new RegExp(
```
Comment (zackify) [inline src/parser/to-js/uri.ts]: maybe use more descriptive variable name, even if that means its long

**Comment by zackify** [comment]
This is really great work, just respond to my couple comments and lets merge it!

File: tests/parser/to-js/uri.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  'host',
+  'port',
+  'params',
+  'headers',
+]);
+
+test('parses uri string to js object', () => {
+  const t1 = parseURI(uris[0]);
+  expect(new Set(Object.keys(t1))).toEqual(expectedKeys);
+  expect(t1).toEqual({
```
Comment (zackify) [inline tests/parser/to-js/uri.js]: might be cleaner and more legible if you make a new test for each type of uri. ex: "parses uri with a branch" "parses uri with just sip address" and then put the uri in that actual test method. currently, its hard to know what is happening or what exactly is being expected in the tests. also, im a big fan of never doing short variables. i would rather have `parsedUri` than `t1`, i dont know what t1 is supposed to mean. lmk if you feel different :D also i forget these things too and its possible i do them still, so catch me if i do haha

File: src/parser/to-js/header.ts (hunk 1 above)
Comment (audiolion) [reply src/parser/to-js/header.ts]: caching the regex basically

File: tests/parser/to-js/uri.js (hunk 3 above)
Comment (audiolion) [reply tests/parser/to-js/uri.js]: Agreed, I had to run so I was trying to get the tests done as fast as possible! t1, t2 was just test 1, test 2. I will go back and make these tests read better.

File: src/parser/to-js/header.ts (hunk 1 above)
Comment (zackify) [reply src/parser/to-js/header.ts]: If you want to cache it, how about giving it a clearer name. I think that or not bothering with the caching is fine, either one. Thoughts?

File: src/parser/to-js/address.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+// sip:bob.castner@example.com
+const addressV3 = /^[\ \t]*()([^;]+)/;
+
+const addresses = [addressV1, addressV2, addressV3];
+
+export default (address: string): Address => {
+  const version = addresses.find(version => {
+    const match = address.match(version);
+    return !!match;
```
Comment (zackify) [inline src/parser/to-js/address.ts]: Does 
Does this not work?
```
addresses.find(version => address.match(version))
```

**Pull request merged**